packages/terra-consumer-nav/src/components/nav-items/NavItem.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './NavItem.scss';
+import NavToggler from '../nav-toggler/NavToggler';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  slug: PropTypes.string,
+  navType: PropTypes.oneOf(['grouping', 'external', 'application', 'modal']),
+  target: PropTypes.string,
+  uri: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+  isActive: PropTypes.bool,
+  badgeValue: PropTypes.number,
+
+  toggleId: PropTypes.number,
+  handleToggle: PropTypes.func,
+  isOpen: PropTypes.bool,
+  /*
+    TODO: Need to figure out what we are doing with icons or if navitems should even have icons
+
+    icon: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.element,
+    ]),
+  */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  isActive: false,
+  handleToggle: () => {},
+};
+
+const NavItem = ({
+  slug,
+  navType,
+  target,
+  uri,
+  text,
+  isActive,
+  badgeValue,
+  toggleId,
+  handleToggle,
+  isOpen,
+  children,
+  ...customProps
+}) => {
+  const activeClass = cx([
+    { active: isActive },
+  ]);
+
+  const itemText = (
+    <span>
+      {text}
+      { badgeValue > 0 &&
+        <span className={cx('badge')}>{badgeValue}</span>
+      }
+    </span>
+  );
+
+  const currentItem = (children && children.length > 0) ?
+    (<NavToggler
+      text={itemText}
+      subItems={children}
+      handleToggle={open => handleToggle(toggleId, open)}
+      isOpen={isOpen}
+      className={cx('nav-item')}
+    />)
+  :
+  (<div className={activeClass}>
+    <a href={uri} className={cx('nav-item')}>
+      {/* When Icon is figured out: <span className={cx('nav-icon')}>{icon}</span> */}
+      {itemText}
+    </a>
+  </div>);
+
+  return (
+    <div key={text} className={cx('nav-item-wrapper')} {...customProps}>
```

Comment:
customProps need to be filtered, there is an error msg on the nightwatch test:

<img width="575" alt="screen shot 2017-08-07 at 1 59 31 pm" src="https://user-images.githubusercontent.com/5943136/29041500-aea05286-7b78-11e7-9666-3a22faab156a.png">